apps/core/client/queries/get-product-search-results.ts

```diff
@@ -173,7 +176,23 @@ export const getProductSearchResults = cache(
 
     const response = await client.fetch({
       document: query,
-      variables: { first: limit, after, filters, sort, imageHeight, imageWidth },
+      variables: before
+        ? {
+            before,
+            last: limit,
+            filters,
+            sort,
+            imageHeight,
+            imageWidth,
+          }
+        : {
+            first: limit,
+            after,
+            filters,
+            sort,
+            imageHeight,
+            imageWidth,
+          },
```

Comment:
Any recommendations on how to do this? We can't mix `last` and `first` in the same query.

Reply:
We do this in multiple places in the repo, check `getBlogPosts` for eg.